### PR TITLE
Trm 22884 advanced search xref

### PR DESCRIPTION
**Pull request by dlrice**
- [x] What is the link to the Jira that this PR is for?
https://www.ebi.ac.uk/panda/jira/browse/TRM-22884
- [x] Have you written tests and do these have >= 75% coverage of the code that you are contributing?
94.74% of lines
- [x] Is it possible to visually evaluate this PR in the browser? If so, what are the steps to do this (eg specific URLs or sections)?
Yes, query builder > search for embl > select "Cross-references / Sequence databases / EMBL" > enter "*" for value and click search
- [x] Are there any linting errors that required you to add exceptions?
N/A
- [x] If there are backend requests, have you checked for the existence of the resource's attributes before attempting access?
N/A

### CODE REVIEW

File: src/search/utils/QueryStringGenerator.ts [hunk 1]
```diff
@@ -31,11 +31,15 @@ const createTermString = (
...
   itemType: string,
   id: string | undefined,
-  termSuffix: boolean | undefined
+  termSuffix: boolean | undefined,
+  stringValue: string | undefined
 ) => {
   if (term === undefined) {
     throw new Error('term is undefined');
   }
+  if (term === 'xref' && stringValue === '*') {
```
Comment (xwatkins) [inline src/search/utils/QueryStringGenerator.ts]: I don't know if whitespace is trimmed from `stringValue` anywhere but if not it probably should.

File: src/search/utils/QueryStringGenerator.ts [hunk 2]
```diff
@@ -56,12 +60,22 @@ const createValueString = (
...
   let valueString = stringValue;
-  if (valuePrefix && !(term === 'xref' && valuePrefix === 'any')) {
-    valueString = `${valuePrefix}-${stringValue}`;
+  if (term === 'xref') {
+    if (!valuePrefix) {
+      throw new Error('valuePrefix not provided in xref query');
+    }
+    // The API will run more quickly when all database entries are
+    // requested by the user if xref-X:* becomes database:X
+    else if (term === 'xref' && stringValue === '*') {
```
Comment (xwatkins) [inline src/search/utils/QueryStringGenerator.ts]: you're already checking `term === 'xref'` in the block

Comment (dlrice) [reply src/search/utils/QueryStringGenerator.ts]: good catch

File: src/search/utils/QueryStringGenerator.ts (hunk 1 above)
Comment (dlrice) [reply src/search/utils/QueryStringGenerator.ts]: Totally agree. I've just a commit that does all the trimming at the point of query string generation as the query builder form elements are controlled.

**Comment by xwatkins** [comment]
I'm wondering if for the sake of simplicity it might be better to trim the string in the `UPDATE_INPUT_VALUE` reducer? That way we only have to do it once ever.

**Comment by dlrice** [comment]
> I'm wondering if for the sake of simplicity it might be better to trim the string in the `UPDATE_INPUT_VALUE` reducer? That way we only have to do it once ever.

That would have been much nicer and I did try this but unfortunately because the advanced search/query builder input elements are controlled components doing this will prevent users from entering spaces (eg "homo sapiens") as the value will constantly be trimmed.

**Pull request merged**